src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -58,42 +39,48 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
             VisualStudioWorkspace workspace)
         {
             _lastTimeReported = DateTimeOffset.UtcNow;
-            _resettableDelay = null;
-
-            ResetProgressStatus();
-
             _taskCenterService = (IVsTaskStatusCenterService)taskStatusCenterService;
-
             _options = new TaskHandlerOptions()
             {
                 Title = ServicesVSResources.Running_low_priority_background_processes,
                 ActionsAfterCompletion = CompletionActions.None
             };
 
-            var crawlerService = workspace.Services.GetService<ISolutionCrawlerService>();
+            var crawlerService = workspace.Services.GetRequiredService<ISolutionCrawlerService>();
             var reporter = crawlerService.GetProgressReporter(workspace);
 
-            StartedOrStopped(reporter.InProgress);
+            _solutionCrawlerEventHandler = new SolutionCrawlerEventHandlerWithDelay(reporter, (int)s_minimumInterval.TotalMilliseconds);
+
+            ResetProgressStatus();
+
+            if (reporter.InProgress)
+            {
+                Started();
+            }
+            else
+            {
+                Stopped();
+            }
 
             // no event unsubscription since it will remain alive until VS shutdown
-            reporter.ProgressChanged += OnSolutionCrawlerProgressChanged;
+            _solutionCrawlerEventHandler.UpdateProgress += OnUpdateProgress;
```

Comment:
Is there a tiny race here if the reporter switches InProgress to false between the check above and this subscription?